### PR TITLE
Fixes baked-in node_threshold_total for TreedropdownField

### PR DESCRIPTION
**Pull request by phptek**
node_threshold_total baked-in as 30 in 3.6

### CODE REVIEW

File: _config/config.yml [hunk 1]
```diff
@@ -17,3 +17,5 @@ HTTP:
     must-revalidate: "true"
     no-transform: "true"
   vary: "Cookie, X-Forwarded-Protocol, User-Agent, Accept"
+Hierarchy:
+  node_threshold_total: 30
```
Comment (tractorcow) [inline _config/config.yml]: For default values it's better to put these in private statics. YML is more for overrides and injector configs.

File: forms/TreeDropdownField.php [hunk 2]
```diff
@@ -337,9 +337,8 @@ public function tree(SS_HTTPRequest $request) {
 		}
 
 		// main node leaf limit configuration
-		$nodeCountThreshold = 30;
```
Comment (tractorcow) [inline forms/TreeDropdownField.php]: You may want a different threshold for treedropdown field than you might use for, say, the sitetree hierarchy view.

File: _config/config.yml (hunk 1 above)
Comment (dhensby) [inline _config/config.yml]: this is already set as a private static so shouldn't be changed here

Comment (tractorcow) [inline _config/config.yml]: This is the framework module, why should the framework set a default for it's own class to two different values in two places?

```
private static $node_threshold_total = 50;
```

This PR changes the default to 30. This above 50 is now redundant.

**Comment by phptek** [comment]
@tractorcow simple. I hadn't spotted the private static! I'll amend ASAP.

**Comment by phptek** [comment]
Since re-testing with 3.6-latest, the `TreeDropdownField` "performance" improvement changes that were reverted - means this issue as reported is no longer a problem. Thanks for your feedback, closing PR.

**Comment by dhensby** [comment]
FYI @phptek the performance improvements still live in the 3 branch which means this will become a problem again in 3.7

**Comment by phptek** [comment]
Hmmm what is the merge process regarding the 3 branch? That confused me. Regardless, I'll re-submit the PR

**Comment by dhensby** [comment]
@phptek the way it works is:

Patch branches (X.Y - 3.5, 3.6) are for patches only, only bug fixes should be going into them. Every so often they are then tagged as the latest "patch" release (X.Y.Z - 3.5.2, etc). Changes to these branches are then merged up into the X branch (3, 4, master). Patches in 3.5 will also be merged into 3.6 and then into 3 to ensure all code fixes make their way into all releases.

Minor branches (X - 3, 4, master) are used to receive new (non-breaking) features which are released less frequently as new "minor" releases (X.Y - 3.6, 3.7, etc). These will contain patches from the previous versions along with new features.

**Comment by phptek** [comment]
@dhensby So should feature-branches for bug-fixes for releases 3.6.x be based off the `3.6` branch or the `3.0` branch then?

**Pull request closed without merging**